### PR TITLE
change `git.commit.verbose` to accept "default"

### PR DESCRIPTION
**Pull request by Ryooooooga**
- **PR Description**

ref: https://github.com/jesseduffield/lazygit/pull/2341

change to ignore `git config commit.verbose` if `git.commit.verbose` is `never`.

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go run scripts/cheatsheet/main.go generate`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [x] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [x] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [x] Docs (specifically `docs/Config.md`) have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

### CODE REVIEW

File: docs/Config.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ git:
     useConfig: false
   commit:
     signOff: false
-    verbose: false
+    verbose: default # one of 'default' | true | false
```
Comment (Ryooooooga) [inline docs/Config.md:70]: `default | always | never` might be better.

**Comment by jesseduffield** [comment]
I'm happy with this but do we need to worry about backwards compatibility? iirc this was recently added and not part of a release yet. Is that right?

**Comment by Ryooooooga** [comment]
#2341 was recently merged into master and is not included in any release.

I would like to hear @knutwalker's opinions.

**Comment by jesseduffield** [comment]
Merged!

**Pull request merged**